external/diagnostics_utils/diagnostics_utils/transform.py

```diff
@@ -0,0 +1,216 @@
+"""
+Transforms operate on diagnostic function inputs to adjust data before
+diagnostic values are calculated.
+
+A transform should take in the transform-specific arguments with a diagnostic
+function argument tuple as the final argument and return the adjusted
+diagnostic function arguments.
+"""
+
```

Comment:
Most of this is directly copied from the prognostic diags, with the exception of `_is_3d`, `select_2(3)d_variables`, and `regrid_zdim_to_pressure_levels`. However, the `DiagArg` here has four args, the last being an additional `delp` data array that is passed around so that the vertical dimension can be regridded to pressure levels. I am open to suggestions of how to handle keeping this necessary delp data around if we want to make the `DiagArg` identical to the prog run diags for future refactoring to use a single package.

Reply:
Would be great to avoid this copy paste by refactoring/generalizing the logic of existing prognostic run code. For example the following data structure should support both the offline/online use-cases
```
class DiagArg:
    prediction: xr.Dataset
    target: xr.Dataset
    grid: xr.Dataset
    delp: Optional[xr.DataArray]
```